Content.Server/GameObjects/Components/Items/Storage/StorageFillComponent.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.Interfaces.Serialization;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Items.Storage
+{
+    [RegisterComponent]
+    internal sealed class StorageFillComponent : Component, IMapInit
+    {
+        public override string Name => "StorageFill";
+
+        private List<PrototypeItemData> _contents = new List<PrototypeItemData>();
+
+#pragma warning disable 649
+        [Dependency] private readonly IEntityManager _entityManager;
+#pragma warning restore 649
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _contents, "contents", new List<PrototypeItemData>());
+        }
+
+        void IMapInit.MapInit()
+        {
+            if (_contents.Count == 0)
+            {
+                return;
+            }
+
+            var storage = Owner.GetComponent<IStorageComponent>();
+            var random = IoCManager.Resolve<IRobustRandom>();
+
+            void Spawn(string prototype)
```

Comment:
Can't you just call the one line down where you call Spawn? It's not really necessary to call a new func for that imo.
If it's for explanation you could also just add a comment